tests/src/Integration/Condition/DataIsEmptyTest.php

```diff
@@ -39,17 +39,17 @@ public function setUp() {
    */
   public function testConditionEvaluation() {
     // Test a ComplexDataInterface object.
-    $entity_adapter_empty = $this->prophesize(ComplexDataInterface::class);
-    $entity_adapter_empty->isEmpty()->willReturn(TRUE)->shouldBeCalledTimes(1);
-
-    $this->condition->getContext('data')->setContextData($entity_adapter_empty->reveal());
-    $this->assertTrue($this->condition->evaluate());
-
-    $entity_adapter_full = $this->prophesize(ComplexDataInterface::class);
-    $entity_adapter_full->isEmpty()->willReturn(FALSE)->shouldBeCalledTimes(1);
-
-    $this->condition->getContext('data')->setContextData($entity_adapter_full->reveal());
-    $this->assertFalse($this->condition->evaluate());
+    //$entity_adapter_empty = $this->prophesize(ComplexDataInterface::class);
```

Comment:
Fago,

The problem here is that the mocks for $entity_adapter_empty and $entity_adapter_full are both now invalid.  phpunit puts out errors like these:

1) Drupal\Tests\rules\Integration\Condition\DataIsEmptyTest::testConditionEvaluation
Prophecy\Exception\Call\UnexpectedCallException: Method call:
  - getDataDefinition()
on Double\ComplexDataInterface\P5 was not expected, expected calls were:
  - isEmpty()

If you like, I can look up how to mock getDataDefinition(), but it's not trivial, and there are not currently any examples in the Rules codebase.  Note also that we use Prophecy, while Core uses the standard mock builder, so there are no relevant examples in core either.

So in any case, the mocks have to be changed in some way or the other.